pkg/datapath/linux/config/config.go

```diff
@@ -549,6 +555,63 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 	return fw.Flush()
 }
 
+// vlanFilterMacros generates VLAN_FILTER macros which
+// are written to node_config.h
+func vlanFilterMacros() (string, error) {
```

Comment:
Unfortunatelly I have no good experiance with cilium yet and I think it will take much for me to write such tests... The output of this function is valid C.

Reply:
Would you be up for trying to do at least some minimal tests like configure a specific set of VLAN devices, then run this logic, then check that the output of the function is detecting the correct devices?

There's some fairly basic "create devices" type logic in `pkg/datapath/link/link_test.go`, you would need to go a bit further to create the appropriate VLAN devices. The docs have some examples about how you can run those individual privileged tests to validate the behaviour locally[0].

[0] https://docs.cilium.io/en/latest/contributing/testing/unit/#testing-individual-packages

Reply:
Ok. Thanks. I will do.